jsx/components/CouncilorDetail.jsx

```diff
@@ -3,19 +3,25 @@ import PropTypes from 'prop-types';
 import OrderList from '../components/OrderList.jsx';
 import { connect } from 'react-redux';
 import slugify from '../utils/slugify';
+import { Redirect } from 'react-router-dom';
 
 class CouncilorDetail extends Component {
   render() {
     const { cid } = this.props.match.params;
 
+    if (!this.props.info.councilor.fetched || this.props.info.councilor.fetching) {
+      return <p>Loading…</p>;
+    }
+
     const councilor = this.props.info.councilor.councilors.find((a) => {
       const slug = slugify(a.name);
-      if (slug === cid) {
-        return a;
-      }
-      return null;
+      return (slug === cid);
     });
 
+    if (!councilor) {
+      return <Redirect to="/404" push={true} />
```

Comment:
The `to` destination here is arbitrary; let me know if there's something that makes more sense!

Reply:
I'd love to have it not redirect at all. That way it doesn't throw out the information the visitor had stored in the URL.